### ISSUE TITLE
Install Unable to Connect to DB

### ISSUE THREAD

**Issue by crashland66** [friendica/friendica#9484]
I'm working to get Friendica setup for the first time.  I pulled the stable branch, installed the required modules etc.  Pointed my browser to the new site and got a screen full of green lights, everything required is in place.  I can then move to the next screen, set the https option etc, and then on to the database screen.  

My settings point to localhost, the firefaithfellowship DB, a user of the same name and the password.  This connection info works just fine from the command line, but the installer says that it can't connect.  I can find nothing in the log files to provide any specific information.

If I directly create a local.config.php file with the same DB info, the system fails with a 500 error, indicating that it can't connect to the db I guess, can't find any details in the logs.

I'm installing it on a Debian system with apache.  Any help of where to look would be much appreciated.

**Comment by MrPetovan** [comment]
A 500 error will appear in your Apache server logs as a "PHP Fatal Error", please look over there (`/var/log/apache2/error.log` by default on Debian).

**Comment by crashland66** [comment]
Sadly nothing appears in the apache error log and no php errors show up in the syslog (which is where I send them).  Nothing is there, except and entry in the access log showing a 500 code.  So something in the php code is deciding to return the 500 code because of something that doesn't trigger a php error?

**Comment by MrPetovan** [comment]
This is disappointing but let's go forward, since you pulled the `stable` branch, did you run `bin/composer.phar install --no-dev` in your Friendica folder?

**Comment by crashland66** [comment]
Yes I sure did.  When I point my browser to it, it runs the checks and displays all green.  Everything seems fine until giving the installer the DB info and then it just fails with nothing to indicate why.

**Comment by MrPetovan** [comment]
Ok, can you please temporarily add the following lines at the start your `index.php` file, just after the `<?php` line?
```php
ini_set('display_errors', true);
error_reporting(E_ALL);
```
Does anything show up when you refresh the page?

**Comment by crashland66** [comment]
I'd already turned up all of the error reporting in the php.ini, I added this to the index.php file and unfortunately it still reports absolutely nothing. :(

**Comment by crashland66** [comment]
After running the composer install and pointing to the web server:
]
<img width="866" alt="Screen Shot 2020-11-02 at 12 32 45 PM" src="https://user-images.githubusercontent.com/28628248/97900247-6799ab00-1d08-11eb-9904-07b585b4496f.png">

Everything green and should be good to go, on to the next screen:

<img width="871" alt="Screen Shot 2020-11-02 at 12 33 16 PM" src="https://user-images.githubusercontent.com/28628248/97900373-8c8e1e00-1d08-11eb-929c-32ed00f8afc0.png">

Now to enter the database information and here is the result:

<img width="868" alt="Screen Shot 2020-11-02 at 12 34 11 PM" src="https://user-images.githubusercontent.com/28628248/97900443-a29bde80-1d08-11eb-8488-8a59ad5a58ea.png">

I've tried to just add the right info to the config/local.config.php and doing so results in a 500 error, which is probably whatever is causing the install script to fail.

**Comment by MrPetovan** [comment]
Can you confirm that the PDO module is enabled for both CLI and the PHP Apache module? You can see this information by adding `phpinfo()` to the `index.php` file, and with `php -i` in the CLI.

**Comment by crashland66** [comment]
It's there:

<img width="973" alt="Screen Shot 2020-11-02 at 12 51 14 PM" src="https://user-images.githubusercontent.com/28628248/97901543-56ea3480-1d0a-11eb-995c-0f230883f678.png">

<img width="370" alt="Screen Shot 2020-11-02 at 12 52 20 PM" src="https://user-images.githubusercontent.com/28628248/97901731-8ef17780-1d0a-11eb-94d5-4101fa08b151.png">

**Comment by MrPetovan** [comment]
Having no error message makes it way harder to troubleshoot. Can you please try to change the Apache error log location at least for this virtual host to make sure it isn't a shenanigan with syslog?

**Comment by crashland66** [comment]
Apache logs to it's own logs in /var/log/apache2, with the local.config.php in place the 500 code shows up in the access log but there is nothing in the error log unfortunately.

**Comment by MrPetovan** [comment]
I thought you said it was logging to the syslog, that's why I suggested to use a specific file instead. Without an error message I have ran out of avenues to explore for now, I hope that @AlfredSK or @annando can take over.

**Comment by crashland66** [comment]
Thanks for trying to help.  It's really a weird way to fail.

**Comment by MrPetovan** [comment]
Wait, during development I've had message-less 500 errors because of a PHP script parse error. Maybe your `local.config.php` isn't correctly formatted. Can you please post your current file with the sensitive information blanked? Maybe your password is tripping the way we're interpolating in the config file during install and results in a faulty PHP config file.

**Comment by crashland66** [comment]
Here's a local.config.php that will fail with the errorless 500.  When trying to run the web installer it simply fails with a can't connect message but never writes out the config file.

```php
<?php
// Local configuration

/* If automatic system installation fails:
 *
 * Copy this file to local.config.php
 *
 * Why local.config.php? Because it contains sensitive information which could
 * give somebody complete control of your database. Apache's default
 * configuration will interpret any .php file as a script and won't show the values
 *
 * Then set the following for your MySQL installation
 *
 * If you're unsure about what any of the config keys below do, please check the static/defaults.config.php file for
 * detailed documentation of their data type and behavior.
 */

return [
	'database' => [
		'hostname' => 'localhost',
		'username' => 'firefaithfellowship',
		'password' => 'xxxxxxxxxxxxxxx',
		'database' => 'firefaithfellowship',
		'charset' => 'utf8mb4',
	],

	// ****************************************************************
	// The configuration below will be overruled by the admin panel.
	// Changes made below will only have an effect if the database does
	// not contain any configuration for the friendica system.
	// ****************************************************************

	'config' => [
		'admin_email' => 'greg@gregstoltz.com',
		'sitename' => 'Fire Faith Fellowship Social Network',
		'register_policy' => \Friendica\Module\Register::OPEN,
		'register_text' => '',
	],
	'system' => [
		'default_timezone' => 'UTC',
		'language' => 'en',
	],
];
```

**Comment by MrPetovan** [comment]
Is there a PHP opening tag `<?php` at the start of this file?

**Comment by crashland66** [comment]
yes, very first line missed it in the copy

<?php

// Local configuration

/* If automatic system installation fails:
 *
 * Copy this file to local.config.php
 *

**Comment by MrPetovan** [comment]
Can you please replace `\Friendica\Module\Register::OPEN` with `'open'` just to be sure it isn't an autoloading issue?

**Comment by crashland66** [comment]
same result.

**Comment by MrPetovan** [comment]
Does your password contain an apostrophe by any chance?

**Comment by crashland66** [comment]
No.  Nothing but alphas in the password.

**Comment by MrPetovan** [comment]
Thanks. What about `ini_set('display_startup_errors', 1);` ?

**Comment by crashland66** [comment]
sadly still nothing displayed but:
Internal Server Error

Apologies but the website is unavailable at the moment.

**Comment by MrPetovan** [comment]
Ok thanks, I'm completely out of ideas now.

**Comment by AlfredSK** [comment]
Are you sure your DB user has all rights for your Friendica DB? I mean 'dbuser'@'localhost' has all the rights? DB rights can be tricky.

**Comment by crashland66** [comment]
Alfred thanks!  I swapped in root and the root password for mysql and the error went away.  Very strange since the user I created for the install has all privileges to the Friendica db except for the grant privilege.

**Comment by MrPetovan** [comment]
Can you please verify it has CREATE TABLE and CREATE VIEW privileges?

Also I just realized that there is a red bubble on the page where you entered your database credentials that says "Couldn't connect to the database".

**Comment by AlfredSK** [comment]
Try ``GRANT ALL PRIVILEGES ON database_name.* TO 'database_user'@'localhost';``

and then `FLUSH PRIVILEGES;`

**Comment by crashland66** [comment]
So with the installation failing.  I created a user and a db of the same name.  The user has all privileges on the db.  This doesn't work.  I granted the user all privileges everywhere except for grant.  That didn't work.  Yes, I flushed the mysql privileges after each change.  The only way I was able to get the install to proceed was to use my mysql root user, which is not the best answer.

**Comment by MrPetovan** [comment]
How do you grant all privileges except for grant?

**Comment by crashland66** [comment]
That's what grant all privileges appears to do automatically.  When I granted everything and looked at the privileges in phpmyadmin it was everything but grant.

**Comment by MrPetovan** [comment]
Ok, what about your MySQL log, anything noteworthy?

**Comment by AlfredSK** [comment]
Is it really the `...@'localhost'` user? In MySQL its a difference if the user e.g. is created as `...@'localhost'` or as `...@'127.0.0.1'`. If it works with root but not with your created db user it is a privileges/rights problem.

If nothing helps I'd recommend to start from scratch. Delete the user, drop the db. Then...

`CREATE DATABASE friendica;`

`CREATE USER 'friendica_adm'@'localhost' IDENTIFIED BY 'password1234';`

`GRANT ALL PRIVILEGES ON friendica.* TO 'friendica_adm'@'localhost';`

`FLUSH PRIVILEGES;`
